### PR TITLE
Nightlite compliance extensions

### LINKED ISSUE #46: Regulatory compliance support
Addition of blacklisting of addresses and addition of encrypted transaction data is expected to make it much easier to meet the needs of regulators.

### PR DESCRIPTION
**Pull request by Westlad**
closes #46

### Description

This PR integrates extensions to support regulatory compliance (blacklisting and encryption).  These extensions are selectable with the `COMPLIANCE` environment variable.  Thus, one can select either conventional (SHA-based), MiMC or compliance versions as required.

### QA Instructions

This can be tested by running it against the unit tests in the [Nightfall compliance branch](https://github.com/EYBlockchain/nightfall/tree/westlad/kyc2) repository, by using this version as the Nightlite NPM package that is imported (soon that Nightfall branch will be pulled into master, at which point master will use to this Nightlite version by default.

### Checklist

- [x] I have reviewed the code changes myself
- [x] My code meets all of the acceptance criteria of the ticket it closes.
- [x] I have removed unused code (e.g., `console.logs`, commented out blocks, etc.)
- [x] I have made all necessary changes to the documentation.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Any dependent changes have been merged and published.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,14 +55,37 @@ contract calls will use.
...
-Nightfall, the `./nightfall-generate-trusted-setup` command calls this function for you.
+The `setup/gm17` directory contains the ZoKrates domain-specific language (`.zok`) files that you
+need in order to run the Nightlite functions. The `generateZokratesFiles()` function will read these
+files and complete the trusted setup for you, skipping the `common` folder, containing shared
+functions.
+
+There are actually three sets of `.zok` files, which are selected by using a combination of the
+`HASH-TYPE` and `COMPLIANCE` environment variables. Allowed values are 'sha'|'mimc' and 'true'
+respectively. Setting other than 'true' for the COMPLIANCE variable will select the conventional
+version (either sha or mimic enabled). Use of `HASH_TYPE='mimic'` with `COMPLIANCE='true'` is not
```
Comment (iAmMichaelConnor) [inline README.md:66]: mimic => mimc typo

**Comment by iAmMichaelConnor** [comment]
docker-compose.yml.old appears to be an empty file (GitHub wouldn't let me attach a comment to that file)

File: README.md (hunk 1 above)
Comment (Westlad) [reply README.md]: Fixed in latest commit

**Comment by Westlad** [comment]
Empty docker-compose.yml.old removed.

**Pull request merged**